yarn-project/archiver/src/archiver/archiver.ts

```diff
@@ -268,46 +261,84 @@ export class Archiver implements ArchiveSource {
     );
   }
 
-  private async updateLastProvenL2Block(provenSynchedTo: bigint, currentL1BlockNumber: bigint) {
-    if (currentL1BlockNumber <= provenSynchedTo) {
-      return;
-    }
-
-    const provenBlockNumber = await this.rollup.read.getProvenBlockNumber();
-    if (provenBlockNumber) {
-      await this.store.setProvenL2BlockNumber({
-        retrievedData: Number(provenBlockNumber),
-        lastProcessedL1BlockNumber: currentL1BlockNumber,
-      });
-    }
-  }
-
   private async handleL2blocks(blockUntilSynced: boolean, blocksSynchedTo: bigint, currentL1BlockNumber: bigint) {
```

Comment:
It's really hard to follow all the logic branches in this function.

Reply:
Will look at clearing it up.